components/PoisList/CategorySelector.vue

```diff
@@ -44,42 +44,35 @@ export default defineNuxtComponent({
 
   computed: {
     menuEntries(): Array<{ value: number, title: string, category: ApiMenuCategory['category'] } | undefined> {
-      const menuIndex: { [key: number]: MenuItem } = {}
-      this.menuItems
-        .filter(menuItem => !menuItem.hidden)
-        .forEach((menuItem) => {
-          menuIndex[menuItem.id] = menuItem
-        })
-
       const locales = this.$i18n.locales
       const localeCompareOptions = locales.map(
         (locale: string | LocaleObject) =>
           typeof locale === 'string' ? locale : locale.code,
       )
 
-      return this.menuItems.filter(
+      return Object.values(this.menuItems).filter(
         menuItem => menuItem.category && !menuItem.hidden,
       )
         .map((menuItem) => {
           const parents: string[] = []
           let parentId = menuItem.parent_id
           while (parentId) {
-            if (!menuIndex[parentId])
+            if (!this.menuItems[parentId])
               return undefined
 
-            const name = menuIndex[parentId].menu_group?.name.fr
-            if (name && menuIndex[parentId].parent_id)
+            const name = this.menuItems[parentId].menu_group?.name.fr
+            if (name && this.menuItems[parentId].parent_id)
               parents.push(name)
```

Comment:
We could `Array.unshift` here, it should allow us to avoid the `Array.reverse` later no ?